src/DistributedArrays.jl

```diff
@@ -880,13 +880,37 @@ for f in (:+, :-, :div, :mod, :rem, :&, :|, :$)
             B = samedist(A, B)
             map_localparts($f, A, B)
         end
+        function ($f){T}(A::DArray{T}, B::Array{T})
+            size(A) == size(B) || throw(DimensionMismatch("first array has size $(size(A)) but second array has $(size(B))"))
+            return DArray(A) do I
+                ($f)(localpart(A), B[I...])
```

Comment:
@amitmurthy Thanks. I wondered if we might also be able to simplify this be reorganizing the constructors. @vchuravy might look into that this afternoon. I was wondering why `construct_darray` wasn't just a `DArray` constructor. Would it be a problem to rename it?